src/algebra/module/composition_series.lean

```diff
@@ -0,0 +1,107 @@
+import order.jordan_holder
+import ring_theory.simple_module
+
+namespace composition_series
+
+namespace lattice
+
+variables {α : Type*} [lattice α]
+
+structure is_maximal (x z : α) : Prop :=
+  (lt : x < z)
+  (bot_or_top : ∀ {y}, x ≤ y → y ≤ z → (y = x ∨ y = z))
```

Comment:
I think this is the same notion as `covby`; perhaps @YaelDillies can confirm

Reply:
Yep, this is exactly `x ⋖ z` (import `order.cover`).

Reply:
Thanks, I will try to replace is_maximal by covby tomorrow!

Reply:
I have a version now which uses covby. I am not sure where to leave the additional lemmas, nor what their correct name should be.